kura/org.eclipse.kura.wire.component.provider/src/main/java/org/eclipse/kura/internal/wire/store/DbWireRecordStore.java

```diff
@@ -224,8 +224,8 @@ private void truncate(final int noOfRecordsToKeep) {
                     this.dbHelper.execute(MessageFormat.format(SQL_TRUNCATE_TABLE, sqlTableName));
                 } else {
                     logger.info(message.partiallyEmptyingTable(sqlTableName));
-                    this.dbHelper
-                            .execute(MessageFormat.format(SQL_DELETE_RANGE_TABLE, sqlTableName, noOfRecordsToKeep));
+                    this.dbHelper.execute(MessageFormat.format(SQL_DELETE_RANGE_TABLE, sqlTableName,
+                            Integer.toString(noOfRecordsToKeep)));
```

Comment:
Basically `MessageFormat.format()` by default prints integers with commas as thousands separators, for example running`MessageFormat.format("{0}", 20000)` will return the string `20,000`.

So in this case if `noOfRecordsToKeep` is greater or equal than 1000 an illegal comma will be placed in the query and it won't compile. This causes the DB store cleanup to fail.